index.js

```diff
@@ -392,27 +417,42 @@ class ClientServiceBase {
     return this._executeContract(request);
   }
 
+  /**
+   * @param {ContractExecutionResult} request
+   * @return {Promise<ContractExecutionResult|void|*>}
+   * @throws {ClientError|Error}
+   */
   async _executeContract(request) {
-    await this._registerToAuditorRequestProof(request);
-
+    const ordered = await this._executeOrdering(request);
     const promise = new Promise((resolve, reject) => {
       this.ledgerClient.executeContract(
-          request,
+          ordered,
           this.metadata,
           async (err, response) => {
             if (err) {
-              reject(err);
-            } else {
-              const isConsistent = await this._executeInAuditor(response);
+              return reject(err);
+            }
+
+            try {
+              const isConsistent = await this._executeValidation(
```

Comment:
@jnmt 
Jun, For your information.
I revised the code here to use `try catch` statement because function `_executeValidation` might return rejects when errors occur from calling `auditorClient.validateExecution`.

Reply:
@supl Good catch! Thank you!